commands/command_checkout.go

```diff
@@ -43,7 +43,12 @@ func checkoutCommand(cmd *cobra.Command, args []string) {
 		pointers = append(pointers, p)
 	})
 
-	chgitscanner.Filter = filepathfilter.New(rootedPaths(args), nil)
+	include, exclude := getIncludeExcludeArgs(cmd)
+	if include != nil || exclude != nil {
```

Comment:
What do you think should happen in the following case?

```ShellSession
$ git lfs checkout --include=foo.bar baz.quux
```

Right now, this change will silently ignore `bar.quux`, and instead build a filter from the `--include` flag only. I am not opposed to this, but I think that there are a few options we should consider first:

1. __Ignore arguments when either `--include` or `--exclude` is given__. This seems like an OK option to me, but I think that we should `Exit()` when arguments are given in this case, something like:

    ```ShellSession
    $ git lfs checkout --include=foo.bar baz.quux
    fatal: arguments not compatible with --include, --exclude
    ```

2. __Ignore `--include` and `--exclude` when arguments are given__. This is a similarly OK option for me, but I would like to do the same sort of warning as above in this case.

3. __Combine command-line arguments with `--include`, `--exclude`__. This is my preferred option. I would like to make the following two commands synonyms of each other:

    ```ShellSession
    $ git lfs checkout --include=foo.bar baz.quux
    $ git lfs checkout --include=foo.bar;baz.quux
    ```

    This way, we don't have asymmetric clobbering behavior (i.e., in the case where one style is ignored but the other isn't), and we don't silently do something unexpected when users have `lfs.fetch{in,ex}clude` set in their `.gitconfig`'s.

I think that option (3) could be accomplished via something like:

```go
include, exclude := getIncludeExcludeArgs(cmd)
include = append(include, rootedPaths(args)...)
chgitscanner.Filter = filepathfilter.New(include, exclude)
```

(and ideally, the first two lines of that change would go into a function somewhere).

Reply:
After I can get tests going, I'll come back to this comment to try to understand it again. 🤣